src/test/java/org/javarosa/core/model/TriggerableDagTest.java

```diff
@@ -913,10 +913,10 @@ public void relative_paths_in_nested_repeats() throws IOException {
         scenario.createNewRepeat("/data/outer[1]/inner");
 
         assertThat(scenario.answerOf("/data/outer[0]/inner[0]/count"), is(intAnswer(1))); // Should be 3
-        assertThat(scenario.answerOf("/data/outer[0]/inner[1]/count"), is(intAnswer(1))); // Should be 3
-        assertThat(scenario.answerOf("/data/outer[0]/inner[2]/count"), is(intAnswer(1))); // Should be 3
+        assertThat(scenario.answerOf("/data/outer[0]/inner[1]/count"), is(intAnswer(2))); // Should be 3
```

Comment:
When you say "as we would expect" in the commit comment, do you mean "as the implementation suggests" rather than "as is correct?" Count calculations *should* update every time a repeat is added, right? Isn't it the case that on full-form recomputation all the inner counts will be 3 and the outer counts will be 2? And that if you delete the first inner repeat instance the count fields for the remaining instances will be updated to 2 but that if you delete the last inner repeat instance no update will happen?

The great news is that I think the contextualization fix is just right! I'll kick it around my brain a bit more to see if there are other test cases that would be worth adding. But this test still seems super confusing. The way that phase 1 of repeat instance creation is described [in the docs](https://github.com/opendatakit/javarosa/blob/master/dag.md) suggests to me that the counts should all be updated and I'm not really clear on why they're not. I think that should be added either to the docs or to the tests or perhaps both.

Reply:
>  Count calculations should update every time a repeat is added, right?

Yes, precisely. I forgot to mention that I was planning on separating the contextualization fix from the wrong behavior we're seeing in those counts into another issue, so that this fix could be merged.

Before this PR they would compute a 1. Now they compute the count at the time of the node's creation, which is still wrong, but better somehow.

Would you rather go all the way and try to fix the wrong count numbers in this PR?

Reply:
Sounds fine to focus on the contextualization fix here. I haven't had a chance to come back and do an honest review yet! Let's just make sure that there's documentation of the issue of counts in repeat instances before a newly added one not getting re-evaluated counts.